src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -19,15 +19,22 @@ WatchPoint::WatchPoint(const u16string& path, shared_ptr<Inotify> inotify, int w
     , path(path) {
 }
 
-bool WatchPoint::cancel() {
+CancelResult WatchPoint::cancel() {
     if (status == WatchPointStatus::CANCELLED) {
-        return false;
+        return CancelResult::ALREADY_CANCELLED;
     }
     status = WatchPointStatus::CANCELLED;
     if (inotify_rm_watch(inotify->fd, watchDescriptor) != 0) {
-        throw FileWatcherException("Couldn't stop watching", path, errno);
+        switch (errno) {
+            case EINVAL:
```

Comment:
Did you manage to add a test for this problem?

Reply:
Not yet. I wanted to get the fix in, and then add a test for closing watchers while directories are being deleted in #197.